test_communication/test/test_subscriber.cpp

```diff
@@ -141,7 +139,14 @@ int main(int argc, char ** argv)
     rclcpp::shutdown();
     return 1;
   }
-  rclcpp::spin(node);
+
+  try {
+    rclcpp::spin(node);
+  } catch (const std::exception & e) {
```

Comment:
Just to make sure I understand correctly: if we throw a `std::runtime_error` in the callback above, we won't get the corruption, but we'll fail the test, correct?

Assuming that is the case, then don't we need to also look into the root of the test failure?

Reply:
Yeap, the root of the test failure is #359.

Reply:
Ah, I see, I didn't understand before.  OK, then this is fine with me with @dirk-thomas comment fixed.  Thanks for the clarification.